### PR TITLE
Document Adaptive Localization

### LINKED ISSUE #6232: Documentation for adaptive localization
Should include:
* What is adaptive localization?
* When to use adaptive localization
* How to use adaptive localization
* Examples
* Examples illustrating the effect of different thresholds

### PR DESCRIPTION
**Pull request by dafeda**
**Issue**
Resolves #6232 

## Pre review checklist

- [ ] Read through the code changes carefully after finishing work
- [ ] Make sure tests pass locally (after every commit!)
- [ ] Prepare changes in small commits for more convenient review (optional)
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [ ] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tommyod** [comment]
Comment on notebooks.

In `iterative_ensemble_smoother` we use [myst_nb](https://github.com/equinor/iterative_ensemble_smoother/blob/32f436b169d6a872e87cb950309349a47c20eb0c/docs/source/conf.py#L43) for notebook support. We store `ipynb` and convert to `.py` in [conf.py](https://github.com/equinor/iterative_ensemble_smoother/blob/main/docs/source/conf.py).

In a hobby project of mine ([generalized-additive-models](https://github.com/tommyod/generalized-additive-models)) I had success using [bsphinx](https://github.com/tommyod/generalized-additive-models/blob/5c30c0c5605b0cc221b27b6ac843d1fbc587b09c/docs/conf.py#L55) and config for `jupytext` [these lines](https://github.com/tommyod/generalized-additive-models/blob/5c30c0c5605b0cc221b27b6ac843d1fbc587b09c/docs/conf.py#L99). The end result is pretty much the same, I think.

File: docs/getting_started/howto/adaptive_localization.ipynb [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+    "The process involves two key decisions: determining what constitutes adjacent parameters and defining a measure of closeness.\n",
+    "\n",
+    "- **Localization - Step-by-Step:**\n",
+    "    - **Select Adjacent Parameters:** Identify a subset of adjacent parameters. Subset may consist of a single parameter.\n",
+    "    - **Identify Close Observations:** Choose observations that are near this subset.\n",
+    "        Closeness can be defined in various ways.\n",
+    "    - **Update:** Use only close observations in the update step.\n",
+    "\n",
+    "Two things need to be decided.\n",
+    "What are adjacent parameters, and how to measure closeness?\n",
```
Comment (oyvindeide) [inline docs/getting_started/howto/adaptive_localization.ipynb:51]: The first sentence of this block and the last are more or less saying the same?

Comment (dafeda) [reply docs/getting_started/howto/adaptive_localization.ipynb]: Fixed

File: docs/getting_started/howto/adaptive_localization.ipynb [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "# Adaptive Localization\n",
+    "\n",
+    "Parameter estimation using few samples (realizations) leads to sampling errors and spurious correlations (see Section 1.3), which can lead to inaccurately strong updates during the estimation process.\n",
```
Comment (kvashchuka) [inline docs/getting_started/howto/adaptive_localization.ipynb:37]: I do not see section numbering, can we use a link instead?

**Comment by dafeda** [comment]
I've added a section on computational considerations.
I think this is good to go.

**Review by oyvindeide — APPROVED** [review]
Good job!

**Pull request merged**